### PR TITLE
S2U-12 5.1.1.8 Tests & Quizzes: Improving i18n and feedback on calculated questions

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/S2U-12

### CODE REVIEW

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java [hunk 1]
```diff
@@ -2938,19 +2938,17 @@ protected List<String> extractInstructionSegments(String instructions) {
...
-          String[] firstResults = CALCQ_DOUBLE_OPEN_BRACKET.split(feedback); // only works because all variables and calculations are already replaced
-          for (String r:firstResults) {
-              results = (CALCQ_DOUBLE_CLOSE_BRACKET.split(r));
-              segments.addAll(Arrays.asList(results));
-          }
-          if (segments.size() == 1) {
-              // add in the trailing segment
-              segments.add("");
-          }
+      Pattern pattern = Pattern.compile(OPEN_BRACKET + OPEN_BRACKET + ".*?" + CLOSE_BRACKET + CLOSE_BRACKET);
```
Comment (adrianfish) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java:2941]: I'd move this into an init or something. There doesn't seem any point compiling this pattern every time this method is called.

Comment (jesusmmp) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java:2941]: You are right! I'll move it. Thanks!

Comment (jesusmmp) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java]: Solved!

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Sounds like a good fix, thanks Jesus!

**Pull request merged**